jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -469,12 +474,62 @@ def current_datum(self) -> dict:
             all_datums_received=True,
         )
 
-    def _get_fileinfo_path(self, fileinfo: pfs.FileInfo) -> Path:
+    def download(self):
+        if len(self._datum_list) == 0 or len(self._datum_list[self._datum_index].data) == 0:
+            raise ValueError("Attempting to download empty or unmounted datum")
+
+        download_dir = (
+            f"{self._DOWNLOAD_DIR}/datum-{datetime.datetime.now().isoformat()}"
```

Comment:
What's the idea behind timestamping this? I don't think it's an issue, but I'm not sure what the upside is

Reply:
Just a way to create a download directory that's unique from the existing one. It was also useful when I was testing (before I added logic to delete old downloads) to see what was downloaded most recently, when I did it, etc. but now that we're removing old downloads I'm not attached to this if you have a better suggestion.

Reply:
I don't think it's better in anyway, but using the datum ID like matthew suggested might be something to consider. Up to you